### PR TITLE
CNF-13523: nodeptpdevice: Add missing description to spec

### PR DESCRIPTION
**Pull request by leo8a**
Adds missing descriptions to the nodeptpdevice spec to improve a bit the user / admin experience with the operand.

/cc @aneeshkp @jzding @josephdrichard

### CODE REVIEW

**Comment by leo8a** [comment]
PTL to the spec definitions, text may need some adjustment based on your feedback

File: api/v1/nodeptpdevice_types.go [hunk 1]
```diff
@@ -21,47 +21,73 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+//+kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp"
+//+kubebuilder:printcolumn:name="HwConfig",type="string",JSONPath=".status.hwconfig.status"
```
Comment (jzding) [inline api/v1/nodeptpdevice_types.go:27]: HwConfig is an array, not string

File: api/v1/nodeptpdevice_types.go [hunk 2]
```diff
@@ -21,47 +21,73 @@ import (
...
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
-
 }
 
 type PtpDevice struct {
-	Name    string `json:"name,omitempty"`
+	// Name is the name of the PTP device.
+	// It is a unique identifier for the device.
+	// +required
```
Comment (jzding) [inline api/v1/nodeptpdevice_types.go:45]: my understanding is anything `omitempty` should be optional. In this case I think the Name should be required, and we should remove `omitempty`. @aneeshkp ?

File: api/v1/nodeptpdevice_types.go [hunk 3]
```diff
@@ -21,47 +21,73 @@ import (
...
-
 type HwConfig struct {
-	DeviceID string              `json:"deviceID,omitempty"`
-	VendorID string              `json:"vendorID,omitempty"`
-	Failed   bool                `json:"failed,omitempty"`
-	Status   string              `json:"status,omitempty"`
-	Config   *apiextensions.JSON `json:"config,omitempty"`
+	// DeviceID is the unique identifier for the hardware device.
+	// +optional
+	DeviceID string `json:"deviceID,omitempty"`
```
Comment (jzding) [inline api/v1/nodeptpdevice_types.go:57]: I think DeviceID should be required @aneeshkp ?

File: api/v1/nodeptpdevice_types.go [hunk 4]
```diff
@@ -21,47 +21,73 @@ import (
...
+// NodePtpDeviceStatus defines the observed state of NodePtpDevice
+type NodePtpDeviceStatus struct {
 
-// NodePtpDevice is the Schema for the nodeptpdevices API
-type NodePtpDevice struct {
-	metav1.TypeMeta   `json:",inline"`
-	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// PtpDevice represents a PTP device available in the cluster node.
+	// This struct contains information about the device, including its name and profile.
+	// +required
```
Comment (jzding) [inline api/v1/nodeptpdevice_types.go:83]: optional

File: api/v1/nodeptpdevice_types.go (hunk 3 above)
Comment (aneeshkp) [reply api/v1/nodeptpdevice_types.go:57]: All new field should be optional , unless we  updated API . 
and introduce conversion webhooks are needed to convert between versions when necessary.

File: api/v1/nodeptpdevice_types.go (hunk 1 above)
Comment (leo8a) [reply api/v1/nodeptpdevice_types.go:27]: gotcha, will remove the proposed new column

**Comment by leo8a** [comment]
/restest

unrelated ci failures
```
level=error msg=Cluster operator console Available is False with Deployment_InsufficientReplicas::RouteHealth_StatusError: DeploymentAvailable: 0 replicas available for console deployment
level=error msg=RouteHealthAvailable: route not yet available, https://console-openshift-console.apps.ci-op-dg0ixgww-047b3.origin-ci-int-aws.dev.rhcloud.com/ returns '503 Service Unavailable'
```

**Review by josephdrichard — APPROVED** [review]
/lgtm
/approved

**Pull request merged**